language-plutus-core/src/Language/PlutusCore/CBOR.hs

```diff
@@ -161,24 +159,24 @@ instance Serialise ann => Serialise (Version ann) where
     decode = Version <$> decode <*> decode <*> decode <*> decode
 
 instance Serialise ann => Serialise (Kind ann) where
-    encode = cata a where
-        a (TypeF ann)           = encodeConstructorTag 0 <> encode ann
-        a (KindArrowF ann k k') = fold [ encodeConstructorTag 1, encode ann, k , k' ]
+    encode = \case
+        Type ann           -> encodeConstructorTag 0 <> encode ann
+        KindArrow ann k k' -> encodeConstructorTag 1 <> encode ann <> encode k <> encode k'
```

Comment:
I was forced to remove the recursion schemes from the CBOR code because they caused a somewhat subtle bug when you're encoding/decoding lists.  Recall that cborg introduces an intermediate monoidal type `Encoding` and chunks of data are converted to this and then all concatenated into a single thing when you finally want to output the encoded version.  When it's outputting a list, it writes a tag saying "here's the start of a list", then writes the encoded entries one after the other, then writes a tag saying "end of list".  When you use a catamorphism, it maps the encoding function over the list, encoding all the entries, and then **it concatenates all of the `Encoding` objects into a single object, omitting the markers at the beginning and end**.  If you write this to disk and then try to read it back in, it'll fail. This is a case where you really don't want to be functorial!